warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -124,13 +135,60 @@ public void remove() {
     }
     
     public boolean hasNext() {
-        if (null == includeHeads) {
+        if (!isInitialized()) {
             throw new IllegalStateException("initialize() was never called");
         }
         
         return next != null;
     }
     
+    @Override
+    public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean inclusive) throws IOException {
```

Comment:
Was this needed for the maxIvaratorResults logic, or is this just bonus code?

Reply:
This was needed.  It is during the seek that the ivarator will fail, so I have to override the seek method to allow catching it and dropping that term.